nucypher/cli/commands/ursula.py

```diff
@@ -442,3 +451,153 @@ def _pre_launch_warnings(emitter, dev, force):
         emitter.echo(DEVELOPMENT_MODE_WARNING, color='yellow', verbosity=1)
     if force:
         emitter.echo(FORCE_MODE_WARNING, color='yellow', verbosity=1)
+
+
+BACKUP_ENCRYPTION_SETTINGS = {'encryption': pyzipper.WZ_AES, 'nbits': 256}
+WORKER_ARCHIVE_ROOT = "worker"
+KEYSTORE_ARCHIVE_ROOT = "keystore"
+
+
+class BackupCliOptions:
+    __option_name__ = 'backup_options'
+
+    def __init__(self,
+                 worker_path: str,
+                 keystore_path: str,
+                 backup_path: str,
+                 password: str,
+                 overwrite: bool,
+                 ) -> None:
+        self.worker_path = Path(worker_path)
+        self.keystore_path = Path(keystore_path)
+        self.backup_path = Path(backup_path)
+        self.password = password.encode()
+        self.overwrite = overwrite
+
+
+
+def _default_keystore_path():
+    # Reference: https://ethdocs.org/en/latest/account-management.html#manual-backup-restore
+    keystore_paths = {
+        'Windows': Path('%appdata%\Roaming\Ethereum\keystore'),
+        'Linux': Path('.ethereum/keystore'),
+        'Darwin': Path('Library/Ethereum/keystore'),
+    }
+    system = platform.system()
+    if system in keystore_paths.keys():
+        return Path.home() / keystore_paths[system]
+    return Path.home()
+
+
+backup_cli_options = group_options(
+    BackupCliOptions,
+    password=click.option('--password',
+                          help="Enter backup password",
+                          prompt="Enter backup password",
+                          hide_input=True,
+                          confirmation_prompt=True),
+    worker_path=click.option('--worker-path',
+                             help="Path to Ursula worker directory",
+                             prompt="Path to Ursula worker directory",
+                             type=click.Path(),
+                             required=True,
+                             default=DEFAULT_CONFIG_ROOT),
+    keystore_path=click.option('--keystore-path',
+                               help="Path to Ursula keystore directory",
+                               prompt="Path to Ursula keystore directory",
+                               type=click.Path(),
+                               required=True,
+                               default=_default_keystore_path()),
```

Comment:
We could do that like so:
```bash
$ nucypher ursula backup --help
# ...
  --keystore-path PATH            Path to Ursula keystore directory [/home/foo/.ethereum/keystore] [required]
```


Default values are shown in prompt if omitted:
```bash
$ nucypher ursula backup --backup-path ursula_backup.zip # --keystore-path is missing
Path to Ursula keystore directory [/home/piotr/.ethereum/keystore]: # awaits input
```
So it's not possible to just use the default path - you have to manually confirm in interactive mode or specify it as a flag. This is a conscious design decision to make accidental changes harder.

Reply:
👍 

> Is it overly cautious?
No. IMHO it is a good thing.

One thing to consider, however, is that we typically add a `--force` option to forgo any prompting, so that CLI functionality can be scriptable. May be good to do that in this instance.